backend/editor/entries.py

```diff
@@ -44,6 +54,104 @@ def create_node(self, label, entry, main_language_code):
         result = get_current_transaction().run(" ".join(query), params)
         return result
 
+    def parse_taxonomy(self, filename):
+        """
+        Helper function to call the Open Food Facts Python Taxonomy Parser
+        """
+        # Close current transaction to use the session variable in parser
+        get_current_transaction().commit()
+
+        # Create parser object and pass current session to it
+        parser_object = parser.Parser(get_current_session())
+        try:
+            # Parse taxonomy with given file name and branch name
+            parser_object(filename, self.branch_name, self.taxonomy_name)
+            return True
+        except:
+            return TaxonomyParsingError()
+
+    def import_from_github(self, description):
+        """
+        Helper function to import a taxonomy from GitHub
+        """
+        base_url = "https://raw.githubusercontent.com/openfoodfacts/openfoodfacts-server/main/taxonomies/"
+        filename = self.taxonomy_name + '.txt'
+        base_url += filename
+        try:
+            # Downloads and creates taxonomy file in current working directory
+            urllib.request.urlretrieve(base_url, filename)
+            status = self.parse_taxonomy(filename) # Parse the taxonomy
+            self.create_project(description) # Creates a "project node" in neo4j
+            os.remove(filename) # Removes taxonomy file after parsing

```

Comment:
Great, that's really nice to read :-)

However, here is how to use tempfile, (so that you guarantee no race condition):

up on the file
```
import tempfile
```

```suggestion
            filepath = f"{tmpdir}/{filename}"
            with tempfile.TemporaryDirectory(prefix="taxonomy-") as tmpdir:
                # Downloads and creates taxonomy file in current working directory
                urllib.request.urlretrieve(base_url, filepath)
                status = self.parse_taxonomy(filepath) # Parse the taxonomy
                self.create_project(description) # Creates a "project node" in neo4j
```

no need to clean, the directory is removed at the end.

Reply:
Ah, I was creating a tempfile instead of a directory and was facing errors. This is actually great!